docs/source/en/tasks/image_text_to_text.md

```diff
@@ -0,0 +1,208 @@
+
+Assume we have an application that keeps chat history and takes in the new user input, using `TextIteratorStreamer` is very simple. We will preprocess the inputs as usual, initialize `TextIteratorStreamer` to handle the generation in a separate thread and stream the generated text tokens in real-time. We will pass any generation arguments to the streamer as well.
+
+```python
+import time
+from transformers import TextIteratorStreamer
+from threading import Thread
+
+def model_inference(
+    user_prompt,
+    chat_history,
+    max_new_tokens,
+    images
+):
+    user_prompt = {
+          "role": "user",
+          "content": [
+              {"type": "image"},
+              {"type": "text", "text": user_prompt},
+          ]
+      }
+    chat_history.append(user_prompt)
+    streamer = TextIteratorStreamer(
+        processor.tokenizer,
+        skip_prompt=True,
+        timeout=5.0,
+    )
+
+    generation_args = {
+        "max_new_tokens": max_new_tokens,
+        "streamer": streamer,
+        "do_sample": False
+    }
+    
+    prompt = processor.apply_chat_template(chat_history, add_generation_prompt=True)
+    inputs = processor(
+        text=prompt,
+        images=images,
+        return_tensors="pt",
+    )
+    inputs = {k: v.to(device) for k, v in inputs.items()}
+    generation_args.update(inputs)
+
+    thread = Thread(
+        target=model.generate,
+        kwargs=generation_args,
+    )
+    thread.start()
+
+    acc_text = ""
+    for text_token in streamer:
+        time.sleep(0.04)
+        acc_text += text_token
+        if acc_text.endswith("<end_of_utterance>"):
+            acc_text = acc_text[:-18]
+        yield acc_text
+```
+
+Now let's call the `model_inference` function we created and stream the values. 
+
+```python
+generator = model_inference(user_prompt = "And what is in this image?",
+    chat_history = messages,
+    max_new_tokens = 100,
+    images = images)
+
+for value in generator:
+  print(value)
+
+# In
+# In this
+# In this image ...
+``` 
+## Fit models in smaller hardware
+
+Vision language models are often large and needs optimization to fit them in smaller hardware. transformers supports many model quantization libraries, and here we will only show int8 quantization with Quanto. 
+
+First, install dependencies.
+
+```bash
+pip install -U quanto bitsandbytes
+```
+
+To quantize model during loading, we need to first create `QuantoConfig`. We can load the model like usual, but just pass `quantization_config` during model initialization.
+
+```python
+from transformers import Idefics2ForConditionalGeneration, AutoTokenizer, QuantoConfig
+
+model_id = "HuggingFaceM4/idefics2-8b"
+quantization_config = QuantoConfig(weights="int8")
+quantized_model = Idefics2ForConditionalGeneration.from_pretrained(model_id, device_map="cuda", quantization_config=quantization_config)
+```
+
+And that's it, we can use the model the same way with no changes.
+
+## Further Reading
```

Comment:
Hmm, I'm not sure why this second-level header isn't being parsed. Could you maybe try adding a blankline after line 180?